src/Processor/AbstractSearchProcessor.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+/*
+ * This file is part of the RollerworksSearch package.
+ *
+ * (c) Sebastiaan Stok <s.stok@rollerscapes.net>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Rollerworks\Bundle\SearchBundle\Processor;
+
+use Rollerworks\Component\Search\Input\ProcessorConfig;
+use Rollerworks\Component\Search\SearchConditionInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * AbstractCacheSearchProcessor provides the basic logic for all processors.
+ *
+ * @author Sebastiaan Stok <s.stok@rollerscapes.net>
+ */
+abstract class AbstractSearchProcessor implements SearchProcessorInterface
+{
+    /**
+     * @var Request
+     */
+    protected $request;
+
+    /**
+     * @var SearchConditionInterface
+     */
+    protected $searchCondition;
+
+    /**
+     * @var string
+     */
+    protected $searchCode = '';
+
+    /**
+     * @var string
+     */
+    protected $uriPrefix;
+
+    /**
+     * @var ProcessorConfig
+     */
+    protected $config;
+
+    /**
+     * @var array
+     */
+    protected $errors = [];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSearchCode()
+    {
+        return $this->searchCode;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSearchCondition()
+    {
+        return $this->searchCondition;
+    }
+
+    /**
+     * Returns whether the processed result is valid.
+     *
+     * @return bool
+     */
+    public function isValid()
+    {
+        if (count($this->errors) > 0) {
+            return false;
+        }
+
+        return !($this->searchCondition && $this->searchCondition->getValuesGroup()->hasErrors());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getErrors()
+    {
+        return $this->errors;
+    }
+
+    /**
+     * Gets a Request object value from the request.
+     *
+     * @param string          $name      Name of the parameter
+     * @param string          $default   Default value to use when value is missing or invalid
+     * @param bool            $query     Get from the Query instead of the Request (POST)
+     * @param callable|string $validator Callback to validate the value, default is string
+     *
+     * @return mixed
+     */
+    protected function getRequestParam($name, $default = '', $query = true, $validator = 'is_string')
+    {
+        $params = $query ? $this->request->query : $this->request->request;
+
+        if ($this->uriPrefix) {
+            $value = $params->get($this->uriPrefix.'['.$name.']', $default, true);
+        } else {
+            $value = $params->get($name, $default);
+        }
+
+        // Use default when invalid
+        if ($validator && !call_user_func($validator, $value)) {
```

Comment:
This needs to be changed, to only is_string() for `$query === true` so we can also accept array input for an input processor.